src/wasm-lib/tests/executor/cache.rs

```diff
@@ -254,19 +239,13 @@ extrude(4, sketch001)
     .await
     .unwrap();
 
-    let first = result.first().unwrap();
-    let second = result.last().unwrap();
+    let first = result.first().unwrap().2.artifact_commands.len();
+    let second = result.last().unwrap().2.artifact_commands.len();
 
     assert!(
-        first.2.global.artifact_commands.len() < second.2.global.artifact_commands.len(),
+        first < second,
         "Second should have all the artifact commands of the first, plus more. first={:?}, second={:?}",
-        first.2.global.artifact_commands.len(),
-        second.2.global.artifact_commands.len()
-    );
-    assert!(
-        first.2.global.artifact_responses.len() < second.2.global.artifact_responses.len(),
-        "Second should have all the artifact responses of the first, plus more. first={:?}, second={:?}",
-        first.2.global.artifact_responses.len(),
-        second.2.global.artifact_responses.len()
```

Comment:
The assertion on the responses has been lost.

Reply:
what is going onnnn

Reply:
Yeah, that was deliberate - it did seemed like an implementation detail which shouldn't be visible externally and wouldn't fail unless the assertion which is preserved failed or there is an engine problem

Reply:
I added the assertion specifically because I had a bug in the initial draft implementation of the artifact graph where I wasn't correctly moving the responses to exec_state. They don't originate there. It works now. But we have very few tests for behavior across multiple executions where caching comes into play. If you want to delete it, go ahead, but it makes me nervous how few tests exercise this.

Reply:
I'll have a look if there's a way we can test this with a unit test so we don't need to expose that just for testing. Unless you know if there is a proxy for that data which is already exposed?

Reply:
The artifact graph is the main output derived from both commands and responses that TS needs. So maybe check that, and responses can stay internal.